packages/dashboards/addon/controllers/dashboards/dashboard/view.js

```diff
@@ -44,26 +44,36 @@ export default Controller.extend(ReportToWidget, {
      */
     async updateFilter(dashboard, originalFilter, changeSet) {
       const origFilter = originalFilter.serialize();
+      origFilter.dataSource = originalFilter.dimension.source;
       const newFilters = get(dashboard, 'filters')
         .toArray()
-        .map(fil => fil.serialize()); //Native array of serialized filters
+        .map(fil => {
+          const newFil = fil.serialize();
```

Comment:
why doesn't the serialize function keep the data source in the dimension?

Reply:
Because it's not part of the serialization